src/graph/node.js

```diff
@@ -10,6 +10,17 @@ class Node {
     this.visited = false;
     this.graph = graph;
     this.focused = false;
+    this.distance = new Proxy({}, {
+      get(target, distName, receiver) {
```

Comment:
why cant this just be an object, why do we need to make this complicated with js Proxy and Reflect?

Reply:
distance from %1 to %2 need to return the reference when it is inside the set_to blocks -  “set distance from %1 to %2 to 0” and distance from %1 to %2 needs to return the value when it is used for calculation - "set 'new distance' to distance from source node to 'current' + edge weight of 'edge from 'neighbor' to 'current'' ". Besides, the key of this.distance will be the other node.id which is dynamic. Therefore, I make the dynamic setter and getter with proxy

Reply:
I write the test in the node.text.js and you could have a look.

Reply:
Using an object (distance = {}) is sufficient enough, I think there's no need to complicate this solution

Reply:
But if only using an object (distance = {}), I could not set the initial value of distance[node.id]. I would like to keep the distance[node.id] will be the initial value like null or -1.

Reply:
I want to set the initial value for the distance[node.id] because returning the value of distance[node.id] maybe happene before the distance[node.id] has been set. I am sure if we need to ban this operation. If we prevent the user from getting the value of distance[node.id] before setting it, I guess the initial value is not necessary and we could use an object (distance = {})

Reply:
this should all just be encapsulated within the getDistance method imo

Reply:
ie 

```
this.distance = {};

getDistanceToNode(node) {
  if (distance[node.id] === 0) return 0;
  return distance[node.id] || -1;
}
```

no Proxies needed, no reflection needed, simple plain JS that everyone understands

Reply:
I guess it returns the value rather than the reference if you just return distance[node.id].  But we need the distance from %1 to %2 could return the value and the reference.

Reply:
I think there are two main scenarios this block will be used:
1. set `temp` to `distance from `Node a` to `Node b``
in this case we will need the value from the distance object
2. set `distance from `Node a` to `Node b`` to `1`
in this case we will need a so-called `reference`

If you want to prevent returning undefined as a default distance using this current setup, then you probably need to modify ./graph/addNodeEvent.js and set default distances when a new node is created in a graph

Reply:
I guess I could not prevent returning undefined by just adding default distance. Because I need to prevent the distance[node.id] from returning the undefined.  The node.id is dynamic and we could not know the node.id inside the definition of the object distance.

Reply:
have different methods to get and set distance

`
this.distance = {};

getDistanceToNode(node) {
  if (distance[node.id] === 0) return 0;
  return distance[node.id] || infinity;
}

setDistanceToNode(node, dist) {
  distance[node.id] = dist;
}

`

Reply:
Hi Max, I don't think we need the separate setter and getter function now. The getDistance() will directly return the reference of object this. distance to the block. And reading and assigning could be done within the block.